### ISSUE TITLE
work with multiple related records

### ISSUE THREAD

**Issue by lynicidn** [yiisoft/yii2#9510]
am suggest:
1.1 return assigned index after `link` (or with related ['indexed' => $model])
1.2 return revoked index after `unlink`// for unrelated records
or may be move https://github.com/yiisoft/yii2/blob/master/framework/db/BaseActiveRecord.php#L1333-L1337
into method, example:

``` php
function findRelatedByIndex ($name, $index)
{
            foreach ($this->_related[$name] as $a => $b) {
                if ($model->getPrimaryKey() === $b->getPrimaryKey()) {
                    return [$a => $this->_related[$name][$a]];
                }
            }
return null;
}
```

but uncomforted search index of unrelated records (1.2)

**Comment by cebe** [comment]
what do you need this for?

**Comment by lynicidn** [comment]
@cebe 
i think как то так :)
```php
$many = $model->getRelation('many')->indexBy('altname')->all();
ksort($many);
$related = array_shift($many);
$model->unlink('many', $related);
$deletedIndex = array_diff_key($many, $model->many);
```

**Comment by lynicidn** [comment]
not better example with unlink - here can use `foreach`, but if work with link, index key defined after link

**Comment by lynicidn** [comment]
$many = $model->many;
$manyKeys = array_keys($many);
$related = new Many;
$model->link('many', $related);
$index = array_diff($manyKeys, array_keys($model->many));

**Comment by lynicidn** [comment]
https://github.com/yiisoft/yii2/blob/master/framework/db/BaseActiveRecord.php#L1239

**Comment by lynicidn** [comment]
also can as
```php
$indexBy = $this->getRelation('many')->indexBy;
$index = $indexBy ? $model->$indexBy : ???;
```

**Comment by cebe** [comment]
I mean what are you going to do with the index when you have it?

**Comment by lynicidn** [comment]
@cebe in wrapper for relation logic [un]set inner flag :)
example multi form
1 => ['a' => 'a'],
2 => ['a' => 'b'],
'new' => ['a' => 'c'],
after save(link) i should restruct array of tabular input for next update where key of current attributes is pk

**Comment by lynicidn** [comment]
`where key of current attributes is pk`  - it abstract

**Comment by samdark** [comment]
Won't be implemented in 2.0.
